### PR TITLE
Don't send patches through a UTF8 encode/decode step

### PR DESCRIPTION
**Pull request by jgraham**
This will garble any commits that are testing non-UTF8 encodings.

### CODE REVIEW

File: sync/commit.py [hunk 1]
```diff
@@ -160,7 +160,11 @@ def is_empty(self, prefix=None):
             args = ("--", prefix)
         else:
             args = ()
-        return self.repo.git.show(self.sha1, format="", patch=True, *args).strip() == ""
+        return self.repo.git.show(self.sha1,
+                                  format="",
+                                  patch=True,
+                                  stdout_as_string=False,
```
Comment (whimboo) [inline sync/commit.py]: `False` is the default so you can keep the line untouched.

Comment (jgraham) [reply sync/commit.py]: No, `True` is the default, that's the bug that's being fixed here

Comment (whimboo) [reply sync/commit.py]: So what does this line do then? https://github.com/mozilla/wpt-sync/blob/c50d5d75743355c403bead18391a8f8040b85888/sync/commit.py#L187

Comment (jgraham) [reply sync/commit.py]: That code isn't being called in this case.

Comment (whimboo) [reply sync/commit.py]: So this shows my missing knowledge of this code as pointed out 27 days ago. Given that I don't have the time to dive into all this, I trust you and the passing new tests. Lets get it shipped.

**Pull request merged**